src/test/java/com/example/demo/StudentControllerTest/StudentControllerTest.java

```diff
@@ -0,0 +1,128 @@
+package com.example.demo.StudentControllerTest;
+
+
+import com.example.demo.StudentController.StudentController;
+
+import com.example.demo.entity.Student;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+@ExtendWith(MockitoExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class StudentControllerTest {
+
+
+    @InjectMocks
+    StudentController controller;
+    @LocalServerPort
+    int randomServerPort;
+
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
+        controller = new StudentController();
+    }
+
+    @Test
+    void shouldListStudents() throws URISyntaxException {
+        RestTemplate restTemplate = new RestTemplate();
+
+        final String baseUrl = "http://localhost:" + randomServerPort + "/students";
+        URI uri = new URI(baseUrl);
+
+        ResponseEntity<String> result = restTemplate.getForEntity(uri, String.class);
+        //Verify request succeed
+        
+        Assert.assertEquals(200, result.getStatusCodeValue());
+        Assert.assertEquals(true, result.hasBody());
+    }
+
+    @Test
+    void shouldGetStudentById() throws URISyntaxException {
+        RestTemplate restTemplate = new RestTemplate();
+
+        final String baseUrl = "http://localhost:" + randomServerPort + "/students/{id}";
+        try{

```

Comment:
> try should never be used in tests. 
> we would create test data first and then assert the results with the expected outcome.
> there should be separate tests for positive (where data is available and asserts) and negative (where an exception is expected)
> exceptions are also asserted instead of try/catch